api/notd/token_manager.py

```diff
@@ -509,3 +509,43 @@ async def update_latest_listings_for_collection(self, address: str) -> None:
             await self.saver.delete_latest_token_listings(latestTokenListingIds=currentLatestTokenListingIds, connection=connection)
             logging.info(f'Saving {len(allListings)} listings')
             await self.saver.create_latest_token_listings(retrievedTokenListings=allListings, connection=connection)
+
+    async def update_partial_latest_listings_for_collection(self, address: str, startDate: datetime.datetime) -> None:
+        queryData = {
+            'asset_contract_address': address,
+            'occurred_after': startDate,
+        }
+        tokensToReprocess = set()
+        while True:
+            response = await self.tokenListingProcessor.openseaRequester.get(url="https://api.opensea.io/api/v1/events", dataDict=queryData, timeout=30)
+            responseJson = response.json()
+            print(f'Got {len(responseJson["asset_events"])} items')
+            for asset in responseJson['asset_events']:
+                if asset['asset'] and asset.get('event_type') != 'bid_entered':
+                    tokensToReprocess.add(asset['asset']['token_id'])
+            if responseJson['next'] is None:
+                break
+            queryData['cursor'] = responseJson['next']
+            await asyncio.sleep(0.25)
```

Comment:
mvoe this part into the tokenListingProcessor

Reply:
I should create a new function in the tokenListingProcessor

Reply:
yes. something like `get_changed_opensea_listings_for_collection(address: str, startDate: datetime.datetime)` you can move the call to `get_opensea_listings_for_tokens` in there too so the function returns a list of Listings